### DISCUSSION TITLE
Eager loading a collection query to avoid N+1 problems in loops

### DISCUSSION THREAD

**Discussion by aaronbushnell** [statamic/cms#8194] [Q&A] [answered]
This might have been answered elsewhere, but I couldn't quite find it! Is it possible to eager or lazy load collections that call to relationships to avoid the dreaded N+1 problem?

I've tried both of these with no luck — the category query still shows up as a duplicate query:
```php
\Statamic\Facades\Entry::query()
  ->where('collection', 'blog')
  ->with('blog_categories')
  ->paginate(12);

Statamic::tag('collection:blog')
  ->with('blog_categories')
  ->paginate(12)
  ->fetch()
```

**Comment by jasonvarga** [comment, marked as answer]
Not possible at the moment unfortunately, but we're definitely aware of it and want to figure it out at some point.

**Reply by aaronbushnell** [reply]
No problem! Just wanted to make sure I wasn’t missing something. Thank you ☺️
